src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -63,26 +63,64 @@ internal static class ModifierUtils
         }
 
         internal static void ReportDefaultInterfaceImplementationModifiers(
-            LanguageVersion availableVersion,
-            LanguageVersion requiredVersion,
+            bool hasBody,
             DeclarationModifiers modifiers,
             DeclarationModifiers defaultInterfaceImplementationModifiers,
             Location errorLocation,
             DiagnosticBag diagnostics)
         {
-            DeclarationModifiers errorModifiers = modifiers & defaultInterfaceImplementationModifiers;
-            var requiredVersionArgument = new CSharpRequiredLanguageVersion(requiredVersion);
-            var availableVersionArgument = availableVersion.ToDisplayString();
-            while (errorModifiers != DeclarationModifiers.None)
+            if (!hasBody && (modifiers & defaultInterfaceImplementationModifiers) != 0)
             {
-                DeclarationModifiers oneError = errorModifiers & ~(errorModifiers - 1);
-                Debug.Assert(oneError != DeclarationModifiers.None);
-                errorModifiers = errorModifiers & ~oneError;
-                diagnostics.Add(ErrorCode.ERR_DefaultInterfaceImplementationModifier, errorLocation,
-                                ConvertSingleModifierToSyntaxText(oneError),
-                                availableVersionArgument,
-                                requiredVersionArgument);
+                LanguageVersion availableVersion = ((CSharpParseOptions)errorLocation.SourceTree.Options).LanguageVersion;
+                LanguageVersion requiredVersion = MessageID.IDS_DefaultInterfaceImplementation.RequiredVersion();
+                if (availableVersion < requiredVersion)
```

Comment:
For inference of tuple names, I factored this kind of code into LanguageVersionFacts. Consider doing something similar:

```C#

        /// <summary>Inference of tuple element names was added in C# 7.1</summary>
        internal static bool InferTupleElementNames(this LanguageVersion self)
        {
            return self >= LanguageVersion.CSharp7_1; 
        }
```